paasta_tools/utils.py

```diff
@@ -2434,6 +2449,32 @@ def get_ldap_reader_password(self) -> str:
     def get_default_push_groups(self) -> List:
         return self.config_dict.get("default_push_groups", None)
 
+    def get_git_config(self) -> Dict:
+        """Gets git configuration. Includes repo names and their git servers.
+
+        :returns: the git config dict
+        """
+        return self.config_dict.get(
+            "git_config",
+            {
+                "git_user": "git",
+                "repos": {
+                    "yelpsoa-configs": {
+                        "repo_name": "yelpsoa-configs",
+                        "git_server": DEFAULT_SOA_CONFIGS_GIT_URL,
+                    },
+                    "services/.*": {"git_server": DEFAULT_SERVICES_GIT_URL},
+                },
+            },
+        )
+
+    def get_git_repo_config(self, repo_name) -> Dict:
+        """Gets the git configuration for a specific repo.
+
+        :returns: the git config dict for a specific repo.
+        """
+        return self.get_git_config().get("repos", {}).get(repo_name, {})
```

Comment:
Agreed, I think this is confusing and I don't really think we'll need more than these 2 git urls. It might be simpler to just have two methods: `get_services_git_url` and `get_soa_configs_git_url`. 

Those methods can handle the default, so you don't have to do stuff like provide the default in the caller like this:
```
system_config.get_git_repo_config("yelpsoa-configs").get(
        "git_server", DEFAULT_SOA_CONFIGS_GIT_URL,
    )
```

Reply:
I decide to remove the services default, and only replace sysgit usages. I need to timebox and make the PR perfect is taking too long. (i.e. this is no longer an issue)